### PR TITLE
Course activity / platform events

### PR DESCRIPTION
**Pull request by bjrne**
You can now view platform events in the news tab.

### CODE REVIEW

File: xikolo-common/Data/Helper/PlatformEventHelper.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        let dateSort = NSSortDescriptor(key: "created_at", ascending: false)
+        request.sortDescriptors = [dateSort]
+        return request
+    }
+
+    static func syncPlatformEvents() -> Future<[PlatformEvent], XikoloError> {
+        return PlatformEventProvider.getPlatformEvents().flatMap { spinePlatformEvents -> Future<[BaseModel], XikoloError> in
+            let request = getRequest()
+            return SpineModelHelper.syncObjectsFuture(request, spineObjects: spinePlatformEvents, inject: nil, save: true)
+        }.map { (cdPlatformEvents) in
```
Comment (invliD) [inline xikolo-common/Data/Helper/PlatformEventHelper.swift]: No need for parenthesis.

File: xikolo-common/Data/Helper/PlatformEventHelper.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    static func getRequest() -> NSFetchRequest {
+        let request = NSFetchRequest(entityName: "PlatformEvent")
+        let dateSort = NSSortDescriptor(key: "created_at", ascending: false)
+        request.sortDescriptors = [dateSort]
+        return request
+    }
+
+    static func syncPlatformEvents() -> Future<[PlatformEvent], XikoloError> {
+        return PlatformEventProvider.getPlatformEvents().flatMap { spinePlatformEvents -> Future<[BaseModel], XikoloError> in
```
Comment (invliD) [inline xikolo-common/Data/Helper/PlatformEventHelper.swift]: Is the return type in the closure necessary? I usually only put them there if the compiler crashes without.

File: xikolo-ios/ViewControllers/News/NewsViewController.swift [hunk 3]
```diff
@@ -10,4 +10,65 @@ import UIKit
...
+    }
+
+    override func updateUIAfterLoginLogoutAction() {
+        super.updateUIAfterLoginLogoutAction()
+        syncContent()
+    }
+
+    func syncContent() {
+        NewsArticleHelper.syncNewsArticles()
+        PlatformEventHelper.syncPlatformEvents()
```
Comment (invliD) [inline xikolo-ios/ViewControllers/News/NewsViewController.swift]: Do we have to load both even though only one is visible at a time?

File: xikolo-ios/ViewControllers/News/PlatformEventsTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        resultsControllerDelegateImplementation.delegate = self
+        resultsController.delegate = resultsControllerDelegateImplementation
+        tableView.dataSource = resultsControllerDelegateImplementation
+
+        do {
+            try resultsController.performFetch()
+        } catch {
+            // TODO: Error handling.
+        }
+        PlatformEventHelper.syncPlatformEvents()
```
Comment (invliD) [inline xikolo-ios/ViewControllers/News/PlatformEventsTableViewController.swift]: Again?

File: xikolo-ios/Views/PlatformEventCell.swift [hunk 5]
```diff
@@ -16,6 +16,20 @@ class PlatformEventCell: UITableViewCell {
     @IBOutlet weak var dateView: UILabel!
     @IBOutlet weak var timelineView: UIView!
 
+    enum EventTypes : String{
```
Comment (invliD) [inline xikolo-ios/Views/PlatformEventCell.swift]: space

File: xikolo-ios/ViewControllers/News/NewsViewController.swift (hunk 3 above)
Comment (bjrne) [inline xikolo-ios/ViewControllers/News/NewsViewController.swift]: In the case of a login or logout, we do. If the user visits the tab for the first time in this launch of the app, we want to already prepare the platform events so they are there when the user switches views. The reload of news articles is not necessary in this particular scenario. Do we care enough about this single request that it makes sense to change it?

File: xikolo-common/Data/Helper/PlatformEventHelper.swift (hunk 2 above)
Comment (bjrne) [inline xikolo-common/Data/Helper/PlatformEventHelper.swift]: I don't really get what you mean, I can't find another helper that does this in a different way.

File: xikolo-ios/ViewControllers/News/NewsViewController.swift (hunk 3 above)
Comment (bjrne) [inline xikolo-ios/ViewControllers/News/NewsViewController.swift]: I created an issue for that in #102.

Comment (jprberlin) [inline xikolo-ios/ViewControllers/News/NewsViewController.swift]: than i am ok with this MR, can be merged after solving conflicts

**Pull request merged**